src/main/driver_unified.cpp

```diff
@@ -447,7 +447,16 @@ int runCvc4(int argc, char* argv[], Options& opts) {
           continue;
         }
 
-        status = pExecutor->doCommand(cmd);
+        try
+        {
+          status = pExecutor->doCommand(cmd);
+        }
+        catch (CVC4::Exception& exc)
+        {
+          // Make sure that no memory is leaked an reraise the exception.
+          delete cmd;
```

Comment:
Can we switch to `unique_ptr` for `cmd` instead of adding the `delete` here?

Reply:
Yes, we can! :-) See the new commits.